src/subdomains/supporting/payment/services/transaction.service.ts

```diff
@@ -10,37 +15,45 @@ import { TransactionRepository } from '../repositories/transaction.repository';
 export class TransactionService {
   constructor(private readonly repo: TransactionRepository) {}
 
-  async create(dto: CreateTransactionDto): Promise<Transaction | undefined> {
+  async create(
+    dto: CreateTransactionDto,
+    sourceEntity: BankTx | CryptoInput | CheckoutTx | RefReward,
+  ): Promise<Transaction | undefined> {
     const entity = this.repo.create(dto);
 
+    const hash = Util.createHash(
+      `${sourceEntity.id}` + entity.created + entity.sourceType + sourceEntity.created,
+    ).toUpperCase();
+    entity.uid = `${hash.slice(0, 8)}-${hash.slice(8, 16)}-${hash.slice(16, 24)}`;
+
```

Comment:
24 digits seems to be a bit long for me 😁  and do you see some benefit using these dashes?